pypeit/metadata.py

```diff
@@ -174,7 +174,12 @@ def _build(self, files, strict=True, usrdata=None):
         # Build the table
         for idx, ifile in enumerate(_files):
             # User data (for frame type)
-            usr_row = None if usrdata is None else usrdata[idx]
+            if usrdata is None:
+                usr_row = None
+            else:
+                # Check
+                assert os.path.basename(ifile) == usrdata['filename'][idx]
```

Comment:
Another pet peeve:  We shouldn't be putting assertions in production-level code.  They're good for checking that the coding is right, but once it's tested, these should either be removed or replaced with an exception that raises a meaningful error.

Reply:
agreed

Reply:
fixed.